test_conformance/extensions/cl_khr_command_buffer/negative_command_nd_range_kernel.cpp

```diff
@@ -0,0 +1,520 @@
+        {
+            return CL_INVALID_VALUE;
+        }
+        num_elements = static_cast<size_t>(elements);
+
+        error = SetUpKernel();
+        test_error(error, "SetUpKernel failed");
+
+        command_buffer = clCreateCommandBufferKHR(1, &queue, nullptr, &error);
+        test_error(error, "clCreateCommandBufferKHR failed");
+
+        return CL_SUCCESS;
+    }
+
+    cl_int Run() override
+    {
+        cl_int error = clCommandNDRangeKernelKHR(
+            command_buffer, nullptr, nullptr, kernel, 1, nullptr, &num_elements,
+            nullptr, 0, nullptr, nullptr, nullptr);
+
+        test_failure_error_ret(error, CL_INVALID_OPERATION,
+                               "clCommandNDRangeKernelKHR should return "
+                               "CL_INVALID_OPERATION",
+                               TEST_FAIL);
+
+        return CL_SUCCESS;
+    }
+};
+
+// CL_INVALID_OPERATION if the device associated with command_queue does not
+// support CL_COMMAND_BUFFER_CAPABILITY_DEVICE_SIDE_ENQUEUE_KHR and kernel
+// contains a kernel-enqueue call.
+struct CommandNDRangeKernelWithKernelEnqueueCall : public BasicCommandBufferTest
+{
+    using BasicCommandBufferTest::BasicCommandBufferTest;
+
+    bool Skip() override
+    {
+        bool has_device_enqueue = false;
+        bool cl_version_2_0_or_higher = false;
+
+        OpenCLCFeatures features;
+        get_device_cl_c_features(device, features);
+
+        const Version clc_version = get_device_latest_cl_c_version(device);
+        if (clc_version >= Version(3, 0))
+        {
+            cl_std = "-cl-std=CL3.0";
+            has_device_enqueue = features.supports__opencl_c_device_enqueue;
+            cl_version_2_0_or_higher = true;
+        }
+        else if (clc_version >= Version(2, 0) && clc_version < Version(3, 0))
+        {
+            cl_std = "-cl-std=CL2.0";
+            has_device_enqueue = features.supports__opencl_c_device_enqueue;
+            cl_version_2_0_or_higher = true;
+        }
+
+        cl_device_command_buffer_capabilities_khr capabilities;
+        cl_int error =
+            clGetDeviceInfo(device, CL_DEVICE_COMMAND_BUFFER_CAPABILITIES_KHR,
+                            sizeof(capabilities), &capabilities, NULL);
+        test_error(error,
+                   "Unable to query CL_DEVICE_COMMAND_BUFFER_CAPABILITIES_KHR");
+
+        bool device_does_support_enqueue_call =
+            (capabilities
+             & CL_COMMAND_BUFFER_CAPABILITY_DEVICE_SIDE_ENQUEUE_KHR)
+            != 0;
+
+        if (!cl_version_2_0_or_higher || !has_device_enqueue) return true;
+        return device_does_support_enqueue_call;
+    }
+
+    cl_int SetUpKernel() override
+    {
+        cl_int error = CL_SUCCESS;
+
+        const char* kernel_str =
+            R"(
+__kernel void enqueue_call_func() {
+  }
+
+__kernel void enqueue_call_kernel() {
+queue_t def_q = get_default_queue();
+ndrange_t ndrange = ndrange_1D(1);
+enqueue_kernel(def_q, ndrange,
```

Comment:
Think we're missing the `flags` parameter here - https://registry.khronos.org/OpenCL/specs/3.0-unified/html/OpenCL_C.html#built-in-functions-enqueuing-a-kernel